mobile/src/bundles/common/store/actions.ts

```diff
@@ -7,16 +7,49 @@ import {
     type UserDetailsGeneralRequestDto,
     type UserDetailsResponseDto,
 } from '~/bundles/common/types/types';
+import { type FileDto } from '~/bundles/file-upload/types/file-dto.type';
 
+import { EMPTY_FILE_COUNT } from '../constants/constants';
+import { mapFilesToPayload } from '../helpers/map-files-to-payload';
 import { name as sliceName } from './slice';
 
 const createUserDetails = createAsyncThunk<
     UserDetailsResponseDto,
     UserDetailsGeneralCreateRequestDto,
     AsyncThunkConfig
 >(`${sliceName}/createUserDetails`, async (onboardingPayload, { extra }) => {
-    const { commonApi, notifications } = extra;
+    const { commonApi, notifications, fileUploadApi } = extra;
     const { photo, companyLogo, ...payload } = onboardingPayload;
+
+    const files: FileDto[] = [];
+
+    if (photo) {
+        const [extension] = photo.name.split('.').reverse();
+        files.push({
+            extension,
+            file: photo,
+        });
+    }
+
+    if (files.length > EMPTY_FILE_COUNT) {
+        const response = await fileUploadApi.upload({ files });
+        mapFilesToPayload({
+            payload: payload,
+            files: response,
+        });
```

Comment:
do you use somewhere result of this function?

Reply:
It is return payload (updating)
![payload](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/112649713/64b27028-549b-40aa-88b1-b7357352d6e2)

Reply:
It should return the payload, do not modify the passed one! It's bad practice